src/CommandHandler.cxx

```diff
@@ -2,397 +2,407 @@
+        x->setStopTime(StopTime);
+      } else {
+        LOG(Sev::Info, "gracefully stop file with id : {}", JobID);
+        x->stop();
       }
-    }
-
-    if (counter == 0) {
-      LOG(Sev::Warning, "no file with id : {}", job_id);
-    } else if (counter > 1) {
-      LOG(Sev::Warning, "error: multiple files with id : {}", job_id);
+      ++counter;
     }
   }
+  if (counter == 0) {
+    LOG(Sev::Warning, "no file with id : {}", JobID);
+  } else if (counter > 1) {
+    LOG(Sev::Warning, "error: multiple files with id : {}", JobID);
+  }
 }
 
-void CommandHandler::handle(rapidjson::Document const &d) {
-  using std::string;
-  using namespace rapidjson;
-  uint64_t teamid = 0;
-  uint64_t cmd_teamid = 0;
-  if (master) {
-    teamid = master->config.teamid;
+void CommandHandler::handle(std::string const &Command) {
+  using nlohmann::json;
+  json Doc;
+  try {
+    Doc = json::parse(Command);
+  } catch (...) {
+    LOG(Sev::Error, "Can not parse json command: {}", Command);
+    return;
   }
-  if (auto i = get_int(&d, "teamid")) {
-    cmd_teamid = int64_t(i);
+  uint64_t TeamId = 0;
+  uint64_t CommandTeamId = 0;
+  if (MasterPtr) {
+    TeamId = MasterPtr->config.teamid;
   }
-  if (cmd_teamid != teamid) {
+  try {
+    CommandTeamId = Doc.at("teamid").get<int64_t>();
+  } catch (...) {
+    // do nothing
+  }
+  if (CommandTeamId != TeamId) {
     LOG(Sev::Info, "INFO command is for teamid {:016x}, we are {:016x}",
-        cmd_teamid, teamid);
+        CommandTeamId, TeamId);
     return;
   }
 
-  // The ways to give commands will be unified in upcoming PR.
-  if (auto s = get_string(&d, "cmd")) {
-    auto cmd = string(s);
-    if (cmd == "FileWriter_new") {
-      handle_new(d);
+  try {
+    std::string CommandMain = Doc.at("cmd");
+    if (CommandMain == "FileWriter_new") {
+      handleNew(Command);
       return;
     }
-    if (cmd == "FileWriter_exit") {
-      handle_exit(d);
+    if (CommandMain == "FileWriter_exit") {
+      handleExit();
       return;
     }
-    if (cmd == "FileWriter_stop") {
-      handle_stream_master_stop(d);
+    if (CommandMain == "FileWriter_stop") {
+      handleStreamMasterStop(Doc);
       return;
     }
-  }
-
-  if (auto s = get_string(&d, "recv_type")) {
-    auto recv_type = string(s);
-    if (recv_type == "FileWriter") {
-      if (auto s = get_string(&d, "cmd")) {
-        auto cmd = string(s);
-        if (cmd == "file_writer_tasks_clear_all") {
-          handle_file_writer_task_clear_all(d);
+    if (CommandMain == "file_writer_tasks_clear_all") {
+      try {
+        std::string ReceiverType = Doc.at("recv_type");
+        if (ReceiverType == "FileWriter") {
+          handleFileWriterTaskClearAll();
           return;
         }
+      } catch (...) {
+        LOG(Sev::Warning, "Can not extract 'recv_type' from command {}",
+            Command);
       }
     }
+  } catch (...) {
+    LOG(Sev::Warning, "Can not extract 'cmd' from command {}", Command);
```

Comment:
The log messages for missing keys are inconsistent in wording and log level. Maybe it would be helpful to create a small function?
```
void logMissingKey(const std::string &key, const std::string &context) {
  LOG(Sev::Warning, "Missing key {} from {}", key, context);
}
```